### PR TITLE
Add CRDs for Route-Agent Health Checker

### PR DESCRIPTION
**Pull request by aswinsuryan**
Add a CRD for routeagent status. There will be status  CR per route agent.

### CODE REVIEW

File: pkg/apis/submariner.io/v1/types.go [hunk 1]
```diff
@@ -185,6 +185,29 @@ const (
...
+type RouteAgentStatus struct {
+	Version         string           `json:"version"`
+	StatusFailure   string           `json:"statusFailure"`
+	RemoteEndpoints []RemoteEndpoint `json:"remoteEndpoints"`
+}
+
+type RemoteEndpoint struct {
+	Status        ConnectionStatus `json:"status"`
+	StatusMessage string           `json:"statusMessage"`
+	Spec          EndpointSpec     `json:"endpoint"`
```
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]:
```suggestion
	Spec          EndpointSpec     `json:"spec"`
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/apis/submariner.io/v1/types.go (hunk 1 above)
Comment (aswinsuryan) [reply pkg/apis/submariner.io/v1/types.go]: Done.

**Review by tpantelis — APPROVED** [review]

File: pkg/apis/submariner.io/v1/types.go [hunk 2]
```diff
@@ -185,6 +185,29 @@ const (
...
+}
+
+type RemoteEndpoint struct {
+	Status        ConnectionStatus `json:"status"`
+	StatusMessage string           `json:"statusMessage"`
+	Spec          EndpointSpec     `json:"spec"`
+	// +optional
+	LatencyRTT *LatencyRTTSpec `json:"latencyRTT,omitempty"`
+}
+
```
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go:210]: Need RouteAgentList as well.

Comment (aswinsuryan) [reply pkg/apis/submariner.io/v1/types.go:210]: Done.

**Review by vthapar — APPROVED** [review]

**Pull request merged**